Sources/StreamChat/Models/ChatMessage.swift

```diff
@@ -61,8 +61,8 @@ public struct ChatMessage {
     ///
     /// If message is inline reply this property will contain the message quoted by this reply.
     ///
-    public var quotedMessage: ChatMessage? { _quotedMessage }
-    @CoreDataLazy internal var _quotedMessage: ChatMessage?
+    public var quotedMessage: ChatMessage? { _quotedMessage() }
+    let _quotedMessage: () -> ChatMessage?
```

Comment:
Why is this still a closure? Can't this be static data as well?

Reply:
`Value type 'ChatMessage' cannot have a stored property that recursively contains it`
I am not sure if there is way. indirect was enums but for properties, hmm

Reply:
[Here](https://forums.swift.org/t/value-type-state-cannot-have-a-stored-property-that-recursively-contains-it/63622/6) is some discussion about it. Hmm

Reply:
Seems like the workaround I have at the moment is the most simple one. But otherwise it is not possible at the moment to have it like other properties.

Reply:
What I mean, is remove the stored property all together, we shouldn't need to have a stored property here 🤔

Reply:
Can you give me a rough example please. Because it is in the public interface then I can't change it much and since there is the compiler error about recursive types then I am not sure how to refactor it.

Reply:
We shouldn't remove the property btw, it's a public API that's used a lot. As for the implementation details, up to you, the closure looks ok to me.

Reply:
This is what I mean:
```suggestion
    public var quotedMessage: ChatMessage?
```

When creating the ChatMessage, we just do this:
`.init(quotedMessage: try? dto.quotedMessage?.relationshipAsModel(depth: depth))`

But I think we could just have `dto.quotedMessage` only maybe? 🤔

Reply:
Ah I got it, it is a limitation of value types